--- a/common/os_calls.c
+++ b/common/os_calls.c
@@ -574,7 +574,19 @@
                 {
                     break; /* Success */
                 }
+                else
+                {
+                    g_writeln("Host address list empty.");
+                }
             }
+            else
+            {
+                g_writeln("Host name not solved.");
+            }
+        }
+        else
+        {
+            g_writeln("Hostname not found.");
         }
     }
     return res;
