--- a/sesman/session.c
+++ b/sesman/session.c
@@ -459,6 +459,7 @@
         g_sprintf(geometry, "%dx%d", width, height);
         g_sprintf(depth, "%d", bpp);
         g_sprintf(screen, ":%d", display);
+        g_sleep(100);     // wait a little time to allow the display chain to be filled in parent thread
         wmpid = g_fork();
 
         if (wmpid == -1)
@@ -642,48 +643,55 @@
             else /* parent (child sesman)*/
             {
                 wait_for_xserver(display);
-                g_snprintf(text, 255, "%d", display);
-                g_setenv("XRDP_SESSVC_DISPLAY", text, 1);
-                g_snprintf(text, 255, ":%d.0", display);
-                g_setenv("DISPLAY", text, 1);
-                /* new style waiting for clients */
-                session_start_sessvc(xpid, wmpid, data, username, display);
+                if (x_server_running(display))
+                {
+                    g_snprintf(text, 255, "%d", display);
+                    g_setenv("XRDP_SESSVC_DISPLAY", text, 1);
+                    g_snprintf(text, 255, ":%d.0", display);
+                    g_setenv("DISPLAY", text, 1);
+                    /* new style waiting for clients */
+                    session_start_sessvc(xpid, wmpid, data, username, display);
+                }
             }
         }
     }
     else /* parent sesman process */
     {
-        temp->item->pid = pid;
-        temp->item->display = display;
-        temp->item->width = width;
-        temp->item->height = height;
-        temp->item->bpp = bpp;
-        temp->item->data = data;
-        g_strncpy(temp->item->client_ip, client_ip, 255);   /* store client ip data */
-        g_strncpy(temp->item->name, username, 255);
-
-        ltime = g_time1();
-        localtime_r(&ltime, &stime);
-        temp->item->connect_time.year = (tui16)(stime.tm_year + 1900);
-        temp->item->connect_time.month = (tui8)stime.tm_mon;
-        temp->item->connect_time.day = (tui8)stime.tm_mday;
-        temp->item->connect_time.hour = (tui8)stime.tm_hour;
-        temp->item->connect_time.minute = (tui8)stime.tm_min;
-        zero_time(&(temp->item->disconnect_time));
-        zero_time(&(temp->item->idle_time));
-
-        temp->item->type = type;
-        temp->item->status = SESMAN_SESSION_STATUS_ACTIVE;
-
-        /*THREAD-FIX require chain lock */
-        lock_chain_acquire();
-
-        temp->next = g_sessions;
-        g_sessions = temp;
-        g_session_count++;
+        wait_for_xserver(display);
+        if (x_server_running(display))
+        {
+            temp->item->pid = pid;
+            temp->item->display = display;
+            temp->item->width = width;
+            temp->item->height = height;
+            temp->item->bpp = bpp;
+            temp->item->data = data;
+            g_strncpy(temp->item->client_ip, client_ip, 255);   /* store client ip data */
+            g_strncpy(temp->item->name, username, 255);
+
+            ltime = g_time1();
+            localtime_r(&ltime, &stime);
+            temp->item->connect_time.year = (tui16)(stime.tm_year + 1900);
+            temp->item->connect_time.month = (tui8)stime.tm_mon;
+            temp->item->connect_time.day = (tui8)stime.tm_mday;
+            temp->item->connect_time.hour = (tui8)stime.tm_hour;
+            temp->item->connect_time.minute = (tui8)stime.tm_min;
+            zero_time(&(temp->item->disconnect_time));
+            zero_time(&(temp->item->idle_time));
+
+            temp->item->type = type;
+            temp->item->status = SESMAN_SESSION_STATUS_ACTIVE;
+
+            /*THREAD-FIX require chain lock */
+            lock_chain_acquire();
+
+            temp->next = g_sessions;
+            g_sessions = temp;
+            g_session_count++;
 
-        /*THREAD-FIX release chain lock */
-        lock_chain_release();
+            /*THREAD-FIX release chain lock */
+            lock_chain_release();
+        }
     }
 
     return display;
