--- debian/rules	2014-08-30 20:39:11.828196607 +0200
+++ debian/rules	2014-08-30 20:39:58.016095353 +0200
@@ -75,7 +75,7 @@
 	config_backend += --disable-config-udev
 endif
 
-vars = $(shell DEB_BUILD_MAINT_OPTIONS="hardening=+pie" DEB_LDFLAGS_MAINT_APPEND="-Wl,-Bsymbolic" DEB_CPPFLAGS_MAINT_APPEND="-DPRE_RELEASE=0 $(cppflags)" dpkg-buildflags --export=configure) LIBS="$(libs)"
+vars = $(shell DEB_BUILD_MAINT_OPTIONS="hardening=+pie" DEB_CFLAGS_MAINT_APPEND="-march=native" DEB_LDFLAGS_MAINT_APPEND="-Wl,-Bsymbolic" DEB_CPPFLAGS_MAINT_APPEND="-DPRE_RELEASE=0 $(cppflags)" dpkg-buildflags --export=configure) LIBS="$(libs)"
 
 ifneq (,$(filter nocheck, $(DEB_BUILD_OPTIONS)))
 	config_tests = --disable-unit-tests
